src/lib.rs

```diff
@@ -3,12 +3,95 @@
 //! This module provides access to Rust implementations of StreamingFast's protocol buffer definitions,
 //! enabling the encoding and decoding of data for Ethereum blockchain and bstream services.
 
+use thiserror::Error;
+
 /// Module for Ethereum-related data structures and operations.
 /// Currently contains the `.proto` defined [here](https://github.com/streamingfast/firehose-ethereum/blob/d9ec696423c2288db640f00026ae29a6cc4c2121/proto/sf/ethereum/type/v2/type.proto#L9)    
 pub mod ethereum {
     pub mod r#type {
         pub mod v2 {
+            use alloy_primitives::{Address, Bloom, FixedBytes, Uint};
+            use ethportal_api::types::execution::header::Header;
+
+            use crate::StreamingFastProtosError;
+
             include!(concat!(env!("OUT_DIR"), "/sf.ethereum.r#type.v2.rs"));
+
+            impl TryFrom<&Block> for Header {
+                type Error = StreamingFastProtosError;
+
+                fn try_from(block: &Block) -> Result<Self, Self::Error> {
```

Comment:
This change currently affects the usage of `header_from_block` in header_accumulator in this commit: https://github.com/semiotic-ai/header_accumulator/commit/2cc72da9729591d80d99c6638c88cb7268bfa0a4


```
let header = header_from_block(&blocks[0].clone()).unwrap();
// can be substituted to a more concise:
 let header = Header::try_from(&blocks[0].clone()).unwrap();
```